packages/elements-core/src/utils/ref-resolving/resolvedObject.test.ts

```diff
@@ -0,0 +1,148 @@
+import { createResolvedObject, getOriginalObject } from './resolvedObject';
+
+describe('createResolvedObject', () => {
+  it('resolves a reference', () => {
+    const resolvedObject = createResolvedObject({
+      paramaterA: {
+        parameterB: {
+          $ref: '#/bundled/parameterB',
+        },
+      },
+      bundled: {
+        parameterB: 'parameterB value',
+      },
+    });
+
+    expect((resolvedObject as any).paramaterA.parameterB).toBe('parameterB value');
+  });
```

Comment:
It's great to have a test that asserts we can resolve to a primitive, but can we add one that tests the main feature: resolving to an object?